lib/components/AlbumScreen/speed_menu.dart

```diff
@@ -0,0 +1,152 @@
+          children: [
+            Padding(
+                padding: EdgeInsets.symmetric(horizontal: 12.0),
+                child: PresetChips(
+                    type: 'speed',
+                    mainColour: widget.iconColor,
+                    values: presets,
+                    activeValue: currentSpeed,
+                    onPressed: () {
+                      setState(() {
+                        currentSpeed =
+                            FinampSettingsHelper.finampSettings.playbackSpeed;
+                        _textController.text = currentSpeed.toString();
+                      });
+                    })),
+            Padding(
+              padding: EdgeInsets.only(top: 8.0, left: 15.0, right: 15.0),
+              child: Form(
+                key: _formKey,
+                child: Row(
+                  children: [
+                    SimpleButton(
+                      text: AppLocalizations.of(context)!.reset,
+                      icon: TablerIcons.arrow_back_up_double,
+                      iconColor: widget.iconColor,
+                      onPressed: () {
+                        _queueService.setPlaybackSpeed(1.0);
+                        setState(() {
+                          currentSpeed = 1.0;
+                          _textController.text = currentSpeed.toString();
+                        });
+                      },
+                    ),
+                    Expanded(
+                      child: Container(
+                        margin: EdgeInsets.symmetric(horizontal: 20.0),
+                        child: TextFormField(
+                          controller: _textController,
+                          keyboardType: TextInputType.number,
+                          textAlign: TextAlign.center,
+                          decoration: inputFieldDecoration(),
+                          validator: (value) {
+                            if (value == null || value.isEmpty) {
+                              return AppLocalizations.of(context)!.required;
+                            }
+
+                            if (double.tryParse(value) == null) {
+                              return AppLocalizations.of(context)!
+                                  .invalidNumber;
+                            }
+                            return null;
+                          },
+                          onSaved: (value) {
+                            final valueDouble =
+                                (min(max(double.parse(value!), 0), 5) * 100)
+                                        .roundToDouble() /
+                                    100;
+
+                            _textController.text = valueDouble.toString();
+
+                            _queueService.setPlaybackSpeed(valueDouble);
+
+                            setState(() {
+                              currentSpeed = valueDouble;
+                            });
+                          },
```

Comment:
This doesn't seem to be working for me. Clicking the confirm button on the keyboard does not change the speed, it changes only after I hit apply

Reply:
@Chaphasilor Yes, I implemented this like the input for the sleep timer dialog (same thing there, having to click "Apply"), but I will try to change that.
Speaking of which, we might want to redesign the sleep timer settings? After all, the preset chips are usable for any preset types ;).

Reply:
Okay, now pressing the confirm button works too.